compute_endpoint/tests/integration/endpoint/executors/test_gcengine_retries.py

```diff
@@ -0,0 +1,149 @@
+            init_blocks=0,
+            min_blocks=0,
+            max_blocks=1,
+        ),
+        strategy=SimpleStrategy(interval=0.1, max_idletime=0),
+    )
+    queue = Queue()
+    engine.start(endpoint_id=ep_id, run_dir=str(tmp_path), results_passthrough=queue)
+
+    yield engine
+    engine.shutdown()
+
+
+def test_gcengine(gc_engine_with_retries):
+    engine = gc_engine_with_retries
+    task_id = uuid.uuid1()
+    serializer = ComputeSerializer()
+    task_body = ez_pack_function(serializer, double, (3,), {})
+    task_message = messagepack.pack(
+        messagepack.message_types.Task(
+            task_id=task_id, container_id=uuid.uuid1(), task_buffer=task_body
+        )
+    )
+    future = engine.submit(task_id, task_message)
+
+    # Confirm that there are no retries on a normal function
+    packed_result = future.result()
+    result = messagepack.unpack(packed_result)
+    assert result.task_id == task_id
+    assert not result.error_details
+
+    # Confirm error message for ManagerLost
+    task_id = uuid.uuid1()
+    task_body = ez_pack_function(serializer, kill_manager, (), {})
+    task_message = messagepack.pack(
+        messagepack.message_types.Task(
+            task_id=task_id, container_id=uuid.uuid1(), task_buffer=task_body
+        )
+    )
+
+    future = engine.submit(task_id, task_message)
+
+    try:
+        packed_result = future.result()
+    except ManagerLost as e:
+        logging.warning(f"This is the exception : {packed_result}")
+        logging.warning(f"Caught result exception : {packed_result}")
+        logging.warning(f"Exception body : {e}")
+    logging.warning("DEBUG: 2")
+
+    engine.shutdown()
+
+
+def test_success_after_1_fail(gc_engine_with_retries, tmp_path):
+    engine = gc_engine_with_retries
+    fail_count = 1
+    queue = engine.results_passthrough
+    task_id = uuid.uuid1()
+    serializer = ComputeSerializer()
+    task_body = ez_pack_function(
+        serializer, succeed_after_n_runs, (tmp_path,), {"fail_count": fail_count}
+    )
+    task_message = messagepack.pack(
+        messagepack.message_types.Task(
+            task_id=task_id, container_id=uuid.uuid1(), task_buffer=task_body
+        )
+    )
+    engine.submit(task_id, task_message)
+
+    flag = False
+    for _i in range(10):
+        q_msg = queue.get(timeout=2)
+        assert isinstance(q_msg, dict)
+
+        packed_result_q = q_msg["message"]
+        result = messagepack.unpack(packed_result_q)
+        # Handle a sneaky EPStatusReport that popped in ahead of the result
+        if isinstance(result, messagepack.message_types.EPStatusReport):
+            continue
+        elif isinstance(result, messagepack.message_types.Result):
+            assert result.task_id == task_id
+            if result.error_details is None:
+                flag = True
+                break
+            else:
+                logging.warning(f"Got failure notice: {result.error_details}")
```

Comment:
I don't think this should be a warning.  If it's of concern and doesn't match what we expect exactly, then we should fail.  If it's not of concern, then there's no need for the log call at all.